src/wazuh_modules/vulnerability_scanner/testtool/scanner/main.cpp

```diff
@@ -66,6 +66,19 @@ class FakeReportServer
         m_clientSize = sizeof(m_clientAddr);
     }
 
+    /**
+     * @brief Destroy the Fake Report Server object and closes the socket.
+     *
+     */
+    ~FakeReportServer()
+    {
+        constexpr auto INVALID_SOCKET {-1};
+        if (m_socketServer != INVALID_SOCKET)
+        {
+            close(m_socketServer);
+        }
+    }
```

Comment:
I think this leaves a `USE_AFTER_FREE` when  in main.cpp:369 calls `fakeDBServer->stop();`

Maybe a solution could be

```cpp
~FakeReportServer()
    {
        waitForStop();
    }
```

And made a little refactor in the function `waitForStop()`, in case of call the destructor with an invalid socket

```cpp
void waitForStop()
    {
        if (m_fakeServerThread.joinable())
        {
            m_fakeServerThread.join();
        }
        constexpr auto INVALID_SOCKET {-1};
        if (m_socketServer != INVALID_SOCKET)
        {
            close(m_socketServer);
            std::filesystem::remove(m_path);
        }
    }
```

Reply:
Good solution